src/components/SQForm/SQFormMultiValue.js

```diff
@@ -109,8 +110,41 @@ function SQFormMultiValue({
 
   const [inputValue, setInputValue] = React.useState('');
   const [customOptions, setCustomOptions] = React.useState([]);
+  const displayOptions = React.useMemo(() => [...children, ...customOptions], [
+    children,
+    customOptions
+  ]);
+  const previousInitialValue = usePrevious(initialValue);
 
-  const initialValue = [];
+  React.useEffect(() => {
+    // Don't run this effect every time displayOptions changes
+    // This also prevents infinite rerenders
+    if (JSON.stringify(previousInitialValue) === JSON.stringify(initialValue)) {
+      return;
+    }
+
+    const displayValues = displayOptions.reduce((acc, option) => {
+      acc.push(option.value);
+      return acc;
+    }, []);
+
+    const newCustomOptions = initialValue.reduce((acc, value) => {
+      if (displayValues.includes(value)) {
```

Comment:
[question] It looks like we loop over displayOptions in line 126 to create an array, displayValues which we loop over on 132 to see if the value is in the array. It would cut another loop to just check here: 
``` 
if (displayOptions.some(({value: displayValue}) => displayValue === value) 
```

the names are awkward there since values is overloaded. Not sure if I am making any sense?

Reply:
Since all of these, correct me if I am wrong here, are the values that the user has added it probably is 10s not 10,000. So it probably doesn't matter.

Reply:
I thought about that but wouldn't that mean that for every iteration of `reduce` we're be running the `.some` loop resulting in _many_ more iterations per execution?

Reply:
I think some is as bad as includes? Maybe I am wrong.

Reply:
I did a simple test on my machine in chrome and found these results with 100001 items and looking for the last item in the array with .some and .includes. I think the biggest thing with `.some()` is that it uses a callback with generates some overhead.
![Screen Shot 2021-06-18 at 10 52 20 AM](https://user-images.githubusercontent.com/29528409/122587398-6614bd00-d023-11eb-958a-ea7a103ec014.png)

Reply:
My original thought was you could cut the map out. If you used `.some()`. If it is trivial, try your test again first mapping an array of objects in to an array of values. Perhaps you already considered that and these number reflect that?

Reply:
I hadn't done that before. I made those changes and ran the tests. I accidently created an infinite loop though so my chrome tab crashed and couldn't screenshot the results. The includes test _was_ slightly slower, about 0.3 - 0.5ms on average.

Reply:
I haven't read through all of this, but FWIW I've looked into the `includes` vs `some` thing before, and I believe `includes` is significantly faster than `some`. `some` is good when you need to check the value against some other arbitrary value because of the callback. Then if you get into "big data" territory a `for` loop is likely faster than `includes`